Description: Try to search jnidispatch.so library first in /usr/lib/jni
 instead of storageDir. If found in /usr/lib/jni, set
 jna.boot.library.path.
Author: Miguel Landaeta <miguel@miguel.cc>
Author: Damien Raude-Morvan <drazzib@debian.org
Forwarded: no
Last-Update: 2013-01-12

--- a/subprojects/native/src/main/java/org/gradle/internal/nativeplatform/jna/JnaBootPathConfigurer.java
+++ b/subprojects/native/src/main/java/org/gradle/internal/nativeplatform/jna/JnaBootPathConfigurer.java
@@ -39,9 +39,17 @@
      */
     public void configure(File storageDir) {
         String nativePrefix = OperatingSystem.current().getNativePrefix();
-        File tmpDir = new File(storageDir, String.format("jna/%s", nativePrefix));
-        tmpDir.mkdirs();
+        File tmpDir = new File(storageDir, "jni/");
         String jnaLibName = OperatingSystem.current().isMacOsX() ? "libjnidispatch.jnilib" : System.mapLibraryName("jnidispatch");
+
+        // Try to load /usr/lib/jni/jnidispatch.so first before storageDir
+        File usrDir = new File("/usr/lib/jni/");
+        File usrLibFile = new File(usrDir, jnaLibName);
+        if (usrLibFile.isFile()) {
+            System.setProperty("jna.boot.library.path", usrDir.getAbsolutePath());
+            return;
+        }
+
         File libFile = new File(tmpDir, jnaLibName);
         if (!libFile.exists()) {
             String resourceName = "/com/sun/jna/" + nativePrefix + "/" + jnaLibName;
